app/controllers/entities_controller.rb

```diff
@@ -0,0 +1,37 @@
+class EntitiesController < ApplicationController
+  before_action :authenticate_user!
+  def index
+    @group = current_user.groups.find(params[:group_id])
+    @entities = @group.entities
+  end
+
+  def new
+    @group = current_user.groups.find(params[:group_id])
+    @entity = @group.entities.new
+  end
+
+  def create
+    @group = current_user.groups.find(params[:group_id])
+    @entity = current_user.entities.create(entity_params)
+    puts @entity
+    if @entity.save
+      @group_entity = @entity.group_entities.create(group_id: @group.id, entity_id: @entity.id)
+      if @group_entity.save
+        flash[:notice] = 'New transaction created successfully'
+        redirect_to group_entities_path(@group)
+      else
+        flash.now[:alert] = 'Transaction category creation failed'
+        render action: 'new'
+      end
+    else
+      flash.now[:alert] = 'Transaction creation failed'
+      render action: 'new'
+    end
+  end
```

Comment:
- [ ] when creating a transaction you should add another input for categories at least once.  so that will make things easy if you want to add transactions to existing categories and also follow the requirements 
- Screenshot 
![Screen Shot 2022-10-11 at 17 00 36](https://user-images.githubusercontent.com/61790587/195142134-9d4bd076-752f-4ff6-97a0-55b547e4c1e2.png)
- Yours 
![Screen Shot 2022-10-11 at 17 00 54](https://user-images.githubusercontent.com/61790587/195142159-0d1a881c-6f93-4fd7-a0c0-943c2e859110.png)